config/confignet/confignet.go

```diff
@@ -5,10 +5,53 @@ package confignet // import "go.opentelemetry.io/collector/config/confignet"
 
 import (
 	"context"
+	"fmt"
 	"net"
 	"time"
 )
 
+// TransportType represents a type of network transport protocol
+type TransportType string
+
+const (
+	TransportTypeTCP        TransportType = "tcp"
+	TransportTypeTCP4       TransportType = "tcp4"
+	TransportTypeTCP6       TransportType = "tcp6"
+	TransportTypeUDP        TransportType = "udp"
+	TransportTypeUDP4       TransportType = "udp4"
+	TransportTypeUDP6       TransportType = "udp6"
+	TransportTypeIP         TransportType = "ip"
+	TransportTypeIP4        TransportType = "ip4"
+	TransportTypeIP6        TransportType = "ip6"
+	TransportTypeUnix       TransportType = "unix"
+	TransportTypeUnixgram   TransportType = "unixgram"
+	TransportTypeUnixPacket TransportType = "unixpacket"
+)
+
+// UnmarshalText unmarshalls text to a TransportType.
+// Valid values are "tcp", "tcp4", "tcp6", "udp", "udp4",
+// "udp6", "ip", "ip4", "ip6", "unix", "unixgram" and "unixpacket"
+func (tt *TransportType) UnmarshalText(in []byte) error {
+	switch typ := TransportType(in); typ {
```

Comment:
What about empty string, that will error.

Reply:
I'll add a `transportTypeEmpty`, like `configcompression` has.